code/game/machinery/slotmachine.dm

```diff
@@ -12,90 +12,94 @@
 	var/resultlvl = null
 
 /obj/machinery/slot_machine/attack_hand(mob/user as mob)
+	tgui_interact(user)
+
+/obj/machinery/slot_machine/tgui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/tgui_state/state = GLOB.tgui_default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "SlotMachine", name, 350, 200, master_ui, state)
+		ui.open()
+		ui.set_autoupdate(TRUE)
```

Comment:
Does the UI need to update this fast? From what I can tell of the comment of `autoupdate`, it updates every MC tick. That seems excessive, but come to think of it, I'm not sure how fast TGUI's update without this var on FALSE.

Reply:
Its every time `SStgui` fires, which is every 0.9 seconds. I think this is required due to the fact the machine needs to update after the spin has completed, which is on a delayed callback. I guess I *could* throw `SStgui.update_uis(src)` into the delayed callback, but this works, and its a very lightweight update.

Reply:
Oh ok, its not that much of a difference then, shouldn't be a big deal.

Reply:
Done it anyway